TestProviders/AzureProvider.psm1

```diff
@@ -67,23 +68,23 @@ Class AzureProvider : TestProvider
 				$customStatus = Set-CustomConfigInVMs -CustomKernel $this.CustomKernel -CustomLIS $this.CustomLIS -EnableSRIOV $enableSRIOV `
 					-AllVMData $allVMData -TestProvider $this
 				if (!$customStatus) {
-					Write-LogErr "Failed to set custom config in VMs, abort the test"
-					return $null
+					$ErrorMessage = "Failed to set custom config in VMs."
+					Write-LogErr $ErrorMessage
+					return @{"VmData" = $null; "Error" = $ErrorMessage}
 				}
 			}
 			else {
 				Write-LogErr "Unable to connect SSH ports.."
 			}
-		}
-		catch {
+		} catch {
 			Write-LogErr "Exception detected. Source : DeployVMs()"
 			$line = $_.InvocationInfo.ScriptLineNumber
 			$script_name = ($_.InvocationInfo.ScriptName).Replace($PWD, ".")
 			$ErrorMessage = $_.Exception.Message
 			Write-LogErr "EXCEPTION : $ErrorMessage"
 			Write-LogErr "Source : Line $line in script $script_name."
 		}
-		return $allVMData
+		return @{"VmData" = $allVMData; "Error" = $ErrorMessage}
```

Comment:
what's the $ErrorMessage here?

Reply:
Just in case that for the future, the implementation changes and there are no returns.

Reply:
acked